# Amazon Seller Sales & Fulfillment Analytics Platform

## 1. Import Libraries

## 2. Load Dataset

## 3. Initial Dataset Exploration

## 4. Data Types

## 5. Missing Values Analysis

## 6. Duplicate Records

## 7. Descriptive Statistics

## 8. Data Dictionary

## 9. Business Observations

## 1. Import Libraries

In [1]:
import pandas as pd
import numpy as np

pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", 100)

## 2. Load Dataset

In [13]:
df = pd.read_csv("../data/raw/amazon_sales.csv")
df.head()

C:\Users\user\AppData\Local\Temp\ipykernel_9084\2198668641.py:1: DtypeWarning: Columns (23) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("../data/raw/amazon_sales.csv")


,index,Order ID,Date,Status,Fulfilment,Sales Channel,ship-service-level,Style,SKU,Category,Size,ASIN,Courier Status,Qty,currency,Amount,ship-city,ship-state,ship-postal-code,ship-country,promotion-ids,B2B,fulfilled-by,Unnamed: 22
0,0,405-8078784-5731545,04-30-22,Cancelled,Merchant,Amazon.in,Standard,SET389,SET389-KR-NP-S,Set,S,B09KXVBD7Z,NaN,0,INR,647.62,MUMBAI,MAHARASHTRA,400081.0,IN,NaN,False,Easy Ship,NaN
1,1,171-9198151-1101146,04-30-22,Shipped - Delivered to Buyer,Merchant,Amazon.in,Standard,JNE3781,JNE3781-KR-XXXL,kurta,3XL,B09K3WFS32,Shipped,1,INR,406.00,BENGALURU,KARNATAKA,560085.0,IN,Amazon PLCC Free-Financing Universal Merchant ...,False,Easy Ship,NaN
2,2,404-0687676-7273146,04-30-22,Shipped,Amazon,Amazon.in,Expedited,JNE3371,JNE3371-KR-XL,kurta,XL,B07WV4JV4D,Shipped,1,INR,329.00,NAVI MUMBAI,MAHARASHTRA,410210.0,IN,IN Core Free Shipping 2015/04/08 23-48-5-108,True,NaN,NaN
3,3,403-9615377-8133951,04-30-22,Cancelled,Merchant,Amazon.in,Standard,J0341,J0341-DR-L,Western Dress,L,B099NRCT7B,NaN,0,INR,753.33,PUDUCHERRY,PUDUCHERRY,605008.0,IN,NaN,False,Easy Ship,NaN
4,4,407-1069790-7240320,04-30-22,Shipped,Amazon,Amazon.in,Expedited,JNE3671,JNE3671-TU-XXXL,Top,3XL,B098714BZP,Shipped,1,INR,574.00,CHENNAI,TAMIL NADU,600073.0,IN,NaN,False,NaN,NaN


## 3. Initial Dataset Exploration

In [14]:
print(f"Rows: {df.shape[0]}")
print(f"Columns: {df.shape[1]}")

Rows: 128975
Columns: 24


In [15]:
df.columns

Index(['index', 'Order ID', 'Date', 'Status', 'Fulfilment', 'Sales Channel ',
       'ship-service-level', 'Style', 'SKU', 'Category', 'Size', 'ASIN',
       'Courier Status', 'Qty', 'currency', 'Amount', 'ship-city',
       'ship-state', 'ship-postal-code', 'ship-country', 'promotion-ids',
       'B2B', 'fulfilled-by', 'Unnamed: 22'],
      dtype='object')

In [ ]:
## 4. Data Types

In [16]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 128975 entries, 0 to 128974
Data columns (total 24 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   index               128975 non-null  int64  
 1   Order ID            128975 non-null  object 
 2   Date                128975 non-null  object 
 3   Status              128975 non-null  object 
 4   Fulfilment          128975 non-null  object 
 5   Sales Channel       128975 non-null  object 
 6   ship-service-level  128975 non-null  object 
 7   Style               128975 non-null  object 
 8   SKU                 128975 non-null  object 
 9   Category            128975 non-null  object 
 10  Size                128975 non-null  object 
 11  ASIN                128975 non-null  object 
 12  Courier Status      122103 non-null  object 
 13  Qty                 128975 non-null  int64  
 14  currency            121180 non-null  object 
 15  Amount              121180 non-nul

## 5. Missing Values Analysis



In [18]:
missing = df.isnull().sum()

missing[missing > 0].sort_values(ascending=False)

fulfilled-by        89698
promotion-ids       49153
Unnamed: 22         49050
currency             7795
Amount               7795
Courier Status       6872
ship-postal-code       33
ship-state             33
ship-city              33
ship-country           33
dtype: int64

## 6. Duplicate Records

In [20]:
print("Duplicate Rows:", df.duplicated().sum())

Duplicate Rows: 0


## 7. Descriptive Statistics

In [24]:
df.describe()

,index,Qty,Amount,ship-postal-code
count,128975.000000,128975.000000,121180.000000,128942.000000
mean,64487.000000,0.904431,648.561465,463966.236509
std,37232.019822,0.313354,281.211687,191476.764941
min,0.000000,0.000000,0.000000,110001.000000
25%,32243.500000,1.000000,449.000000,382421.000000
50%,64487.000000,1.000000,605.000000,500033.000000
75%,96730.500000,1.000000,788.000000,600024.000000
max,128974.000000,15.000000,5584.000000,989898.000000


## 8. Data Dictionary

| Column | Data Type | Description | Business Importance |
|---------|-----------|-------------|---------------------|
| Order ID | Object | Unique identifier for each order | Tracks individual customer orders |
| Date | Date | Date on which the order was placed | Used for time-series and seasonal sales analysis |
| Status | Object | Current order status | Helps analyze delivered, cancelled, and returned orders |
| Fulfilment | Object | Fulfilment method (Amazon or Merchant) | Measures fulfilment performance |
| Sales Channel | Object | Platform through which the order was placed | Evaluates sales source performance |
| ship-service-level | Object | Shipping service level | Measures shipping efficiency |
| Style | Object | Product style/design | Product-level sales analysis |
| SKU | Object | Stock Keeping Unit | Inventory management |
| ASIN | Object | Amazon Standard Identification Number | Unique Amazon product identifier |
| Category | Object | Product category | Identifies top-performing product categories |
| Size | Object | Product size | Inventory and demand planning |
| Courier Status | Object | Shipping progress status | Logistics performance analysis |
| Qty | Integer | Number of items purchased | Sales volume measurement |
| Currency | Object | Currency used in transaction | Financial reporting |
| Amount | Float | Total order value | Revenue analysis |
| ship-city | Object | Customer city | Regional sales analysis |
| ship-state | Object | Customer state | State-wise sales performance |
| ship-postal-code | Float | Customer postal code | Geographic delivery analysis |
| ship-country | Object | Customer country | Country-level sales analysis |
| promotion-ids | Object | Promotion or discount identifier | Marketing campaign analysis |
| B2B | Boolean | Indicates Business-to-Business order | Separates B2B and retail sales |
| fulfilled-by | Object | Fulfilment partner | Supply chain performance |

## Business Observations

- The dataset contains 128,975 Amazon sales records across 24 business attributes.
- Most variables are categorical, reflecting product, customer, and fulfilment information.
- No duplicate records were found, indicating strong data integrity.
- Missing values are primarily concentrated in the fulfilled-by, promotion-ids, Amount, and Currency columns, which will require preprocessing.
- Most customer orders consist of a single item, suggesting predominantly individual purchases.
- Order values show a right-skewed distribution, where a few expensive purchases increase the average order value.
- Geographic information is nearly complete, enabling reliable location-based sales analysis.
- The dataset is suitable for sales performance, fulfilment efficiency, customer behaviour, and regional analytics.